src/main/java/io/ebeaninternal/server/type/RsetDataReader.java

```diff
@@ -31,9 +32,19 @@
 
   protected int pos;
 
+  protected int columnCount;
+  
   public RsetDataReader(DataTimeZone dataTimeZone, ResultSet rset) {
     this.dataTimeZone = dataTimeZone;
     this.rset = rset;
+    if (rset != null) {
+      try {
+        ResultSetMetaData meta = rset.getMetaData();
+        columnCount = meta.getColumnCount();
```

Comment:
Don't rely too on metaData if not really required:
- MsSql driver returns sometimes garbage (getColumnCount should work in most cases)
- metaData can be null in certain cases: e.g. https://github.com/ebean-orm/ebean/blob/94591954b3891d5de6785e72b358fe0d6a7cc582/src/main/java/io/ebeaninternal/server/type/bindcapture/BindCaptureStatementBase.java#L97